cmds/db-manager/main.go

```
@@ -0,0 +1,218 @@
+	*path = fmt.Sprintf("file://%v", *path)
+	migrater, migraterErr := migrate.New(*path, crdbURI)
+	myMigrater := &MyMigrate{migrater}
+	defer func() {
+		if migraterErr == nil {
+			if _, err := myMigrater.Close(); err != nil {
+				log.Println(err)
+			}
+		}
+	}()
+	if migraterErr == nil {
+		myMigrater.PrefetchMigrations = 10
+		myMigrater.LockTimeout = time.Duration(15) * time.Second
+
+		// handle Ctrl+c
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, syscall.SIGINT)
+		go func() {
+			for range signals {
+				log.Println("Stopping after this running migration ...")
+				myMigrater.GracefulStop <- true
+				return
+			}
+		}()
+	}
+	totalMove := 0
+	direction := myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	for direction != 0 {
+		executeErr := myMigrater.Steps(direction)
+		if executeErr != nil {
+			log.Fatal(executeErr)
+		}
+		totalMove += direction
+		log.Printf("Migrated %s by %d step", directionText(direction), int(math.Abs(float64(direction))))
+		direction = myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	}
+	if totalMove == 0 {
+		log.Println("No Changes")
+	} else {
+		log.Printf("Moved %d steps %s in total", int(math.Abs(float64(totalMove))), directionText(totalMove))
+	}
+
+	currentDBVersion, err := getCurrentDBVersion(postgresURI)
+	if err != nil {
+		log.Fatal("Failed to get Current DB version for confirmation")
+	}
+	migrationStep, dirty, err := myMigrater.Version()
+	if err != nil {
+		log.Fatal("Failed to get Migration Step for confirmation")
+	}
+	log.Printf("DB Version: %s, Migration Step # %d, Dirty: %v", currentDBVersion, migrationStep, dirty)
+}
+
+func directionText(direction int) string {
+	if direction > 0 {
+		return "Up"
+	} else if direction < 0 {
+		return "Down"
+	}
+	return "No Change"
+}
+
+func getCurrentDBVersion(crdbURI string) (string, error) {
+	crdb, err := dssCockroach.Dial(crdbURI)
+	if err != nil {
+		log.Panic("Failed to dial CockroachDB")
```

I think you rather want to return the error here.